File: SOLID/DIP/Violation/LibraryService.py
class RegularFineCalculator:
    def calculate_fine(self, late_days):
        return late_days * 5


class LibraryService:
    def __init__(self):
        self._calculator = RegularFineCalculator()

    def get_fine(self, late_days):
        return self._calculator.calculate_fine(late_days)

File: SOLID/DIP/Violation/LibraryService.java
class RegularFineCalculator {
    int calculateFine(int lateDays) {
        return lateDays * 5;
    }
}

class LibraryService {
    private RegularFineCalculator calculator = new RegularFineCalculator();

    int getFine(int lateDays) {
        return calculator.calculateFine(lateDays);
    }
}
